crates/wasmtime/src/profiling.rs

```diff
@@ -0,0 +1,106 @@
+use anyhow::Result;
+use fxprof_processed_profile::{
+    CategoryHandle, CpuDelta, Frame, FrameFlags, FrameInfo, Profile, Timestamp,
+};
+use std::time::{Duration, Instant};
+
+use crate::{AsContext, WasmBacktrace};
+
+// TODO: collect more data
+// - Provide additional hooks for recording host-guest transitions, to be
+//   invoked from a Store::call_hook
+// - On non-Windows, measure thread-local CPU usage between events with
+//   rustix::time::clock_gettime(ClockId::ThreadCPUTime)
+// - Report which wasm module, and maybe instance, each frame came from
+
+// TODO: batch symbolication using Frame::RelativeAddressFromReturnAddress
+
+/// Collects profiling data for a single WebAssembly guest.
+///
+/// To use this, you'll need to arrange to call [`GuestProfiler::sample`] at
+/// regular intervals while the guest is on the stack. The most straightforward
+/// way to do that is to call it from a callback registered with
+/// [`Store::epoch_deadline_callback()`](crate::Store::epoch_deadline_callback).
+///
+/// This implementation requires that the `Engine` was constructed with
+/// [`Config::generate_address_map(true)`](crate::Config::generate_address_map).
+pub struct GuestProfiler {
```

Comment:
I think this'd be a good place to have a doc-block example if you're up for it about how to configure a `Store` and such.